File: 1448-CountGoodNodesinBinaryTree/1448-CountGoodNodesinBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        if (!root){
            return 0;
        }
        
        return 1 + countGN(root->left,root->val) + countGN(root->right, root->val);

        }

    int countGN(TreeNode* node, int max){
        if (!node){
            return 0;

        }
        if (node->val >= max){
            return 1 + countGN(node->left, node->val) + countGN(node->right, node->val);
        }
        
            return countGN(node->left, max) + countGN(node->right, max);
        
    }

};

File: 1448-CountGoodNodesinBinaryTree/1448-CountGoodNodesinBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if not root:
            return 0
        return 1 + self.countGN(root.left, root.val) + self.countGN(root.right, root.val)

    def countGN(self, node: TreeNode, max_val: int) -> int:
        if not node:
            return 0
        if node.val >= max_val:
            return 1 + self.countGN(node.left, node.val) + self.countGN(node.right, node.val)
        return self.countGN(node.left, max_val) + self.countGN(node.right, max_val)